internal/app/singularity/user.go

```diff
@@ -0,0 +1,25 @@
+package singularity
+
+// OidcUserMeta
+type OidcUserMeta struct {
+	ID           int    `json:"id"`
+	UserID       int    `json:"user_id"`
+	Subiss       string `json:"subiss"`
+	Secret       string `json:"secret"`
+	CreationTime string `json:"creation_time"`
+	UpdateTime   string `json:"update_time"`
+}
+
+// User
+type User struct {
+	Email           string       `json:"email"`
+	RealName        string       `json:"realname"`
+	Comment         string       `json:"comment"`
+	UserID          int          `json:"user_id"`
+	UserName        string       `json:"username"`
+	SysAdminFlag    bool         `json:"sysadmin_flag"`
+	AdminRoleInAuth bool         `json:"admin_role_in_auth"`
+	OidcUserMeta    OidcUserMeta `json:"oidc_user_meta"`
+	CreationTime    string       `json:"creation_time"`
+	UpdateTime      string       `json:"update_time"`
+}
```

Comment:
I don't think this is necessary here.. you can include the structs in the `remote_get_login_password.go` and consider making them private as they are only needed for the login-password feature

Reply:
also at this point most of those fields aren't being used, so just unmarshal the fields only used.

Reply:
They are still exposed as public in the singularity package.